graveler/refresolver.go

```diff
@@ -31,6 +32,14 @@ func (r reference) CommitID() CommitID {
 	return *r.commitID
 }
 
+type resolveHelperFunc func(context.Context, RefStore, RepositoryID, string) (Reference, error)
```

Comment:
I like the idea, not sure about the name though. Maybe `TypeResolverFunc`? Or something else that is descriptive

Reply:
refactored with a new name, please re-review this function